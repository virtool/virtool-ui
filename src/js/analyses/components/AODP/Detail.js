import { filter, map, max, mean, min } from "lodash-es";
import React from "react";
import styled from "styled-components";

import { connect } from "react-redux";
import { Badge, BoxGroup, BoxGroupHeader, BoxGroupSection, Icon } from "../../../base";
import { getActiveHit } from "../../selectors";
import { AODPIsolate } from "./Isolate";

const AODPDetailIdentity = styled.div`
    align-items: center;
    display: flex;

    > div:first-child {
        color: ${props => props.theme.color.blue};
        font-size: ${props => props.theme.fontSize.xxl};
    }

    > div:last-child {
        display: flex;
        flex-direction: column;
        font-size: ${props => props.theme.fontSize.sm};
        font-weight: bold;
        justify-content: center;
        margin-left: 5px;

        span:first-child {
            color: ${props => props.theme.color.greenDark};
        }

        span:last-child {
            color: ${props => props.theme.color.redDark};
        }
    }
`;

const AODPDetailOverview = ({ identities }) => {
    return (
        <BoxGroupSection>
            <AODPDetailIdentity>
                <div>{max(identities)}</div>
                <div>
                    <span>~ {mean(identities).toFixed(2)}</span>
                    <span>
                        <Icon name="arrow-down" /> {min(identities)}
                    </span>
                </div>
            </AODPDetailIdentity>
            <div>IDENTITY</div>
        </BoxGroupSection>
    );
};

export const AODPDetail = ({ hit }) => {
    if (hit === null) {
        return <BoxGroup>None found</BoxGroup>;
    }

    const { name, identities, isolates } = hit;

    const filteredIsolates = filter(isolates, isolate => isolate.identities.length);

    const isolateComponents = map(filteredIsolates, isolate => <AODPIsolate key={isolate.id} {...isolate} />);

    return (
        <BoxGroup>
            <BoxGroupHeader>
                <h2>
                    {name} <Badge>{identities.length} hits</Badge>
                </h2>
            </BoxGroupHeader>
            <AODPDetailOverview identities={identities} />
            {isolateComponents}
        </BoxGroup>
    );
};

const mapStateToProps = state => ({
    hit: getActiveHit(state)
});

export default connect(mapStateToProps)(AODPDetail);
