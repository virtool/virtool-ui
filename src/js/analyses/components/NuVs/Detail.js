import { filter, map, sortBy } from "lodash-es";
import React from "react";
import { connect } from "react-redux";
import styled from "styled-components";
import { getBorder } from "../../../app/theme";
import { Badge, Box } from "../../../base";
import { useElementSize } from "../../../utils/hooks";
import { getActiveHit, getMaxSequenceLength } from "../../selectors";
import NuVsBLAST from "./BLAST";
import { NuVsORF } from "./ORF";
import { NuVsSequence } from "./Sequence";
import { NuVsValues } from "./Values";

const StyledNuVsFamilies = styled.div`
    border: ${getBorder};
    border-radius: ${props => props.theme.borderRadius.sm};
    display: flex;
    margin: 10px 0 5px;
    overflow: hidden;

    div {
        padding: 5px 10px;
    }

    div:first-child {
        background-color: ${props => props.theme.color.greyLightest};
        border-right: ${getBorder};
    }
`;

const NuVsFamilies = ({ families }) => (
    <StyledNuVsFamilies>
        <div>Families</div>
        <div>{families.length ? families.join(", ") : "None"}</div>
    </StyledNuVsFamilies>
);

const NuVsLayout = styled.div`
    border: ${getBorder};
    margin-bottom: 15px;

    & > div:nth-child(even) {
        background-color: ${props => props.theme.color.greyLightest};
    }
`;

const NuVsDetailTitle = styled.div`
    margin-bottom: 10px;

    h3 {
        align-items: center;
        display: flex;
        font-size: ${props => props.theme.fontSize.lg};
        font-weight: bold;
        justify-content: space-between;
        margin: 0;
    }

    span {
        font-size: ${props => props.theme.fontSize.md};
        font-weight: bold;
    }

    ${Badge} {
        font-size: ${props => props.theme.fontSize.md};
        padding: 5px 10px;
    }
`;

const StyledNuVsDetail = styled(Box)`
    align-items: stretch;
    display: flex;
    flex-direction: column;
    min-height: 500px;
    min-width: 0;
    margin-left: 10px;
`;

export const NuVsDetail = ({ filterORFs, hit, maxSequenceLength }) => {
    const [ref, { width }] = useElementSize();

    if (!hit) {
        return <StyledNuVsDetail>No Hits</StyledNuVsDetail>;
    }

    const { e, families, index, orfs, sequence } = hit;

    let filtered;

    if (filterORFs) {
        filtered = filter(orfs, orf => orf.hits.length);
    }

    filtered = sortBy(filtered || orfs, orf => orf.hits.length).reverse();

    const orfComponents = map(filtered, (orf, index) => (
        <NuVsORF key={index} index={index} {...orf} maxSequenceLength={maxSequenceLength} width={width} />
    ));

    return (
        <StyledNuVsDetail ref={ref}>
            <NuVsDetailTitle>
                <h3>
                    Sequence {index}
                    <Badge>{sequence.length} bp</Badge>
                </h3>
                <NuVsValues e={e} orfCount={orfs.length} />
                <NuVsFamilies families={families} />
            </NuVsDetailTitle>
            <NuVsLayout>
                <NuVsSequence key="sequence" maxSequenceLength={maxSequenceLength} sequence={sequence} width={width} />
                {orfComponents}
            </NuVsLayout>
            <NuVsBLAST />
        </StyledNuVsDetail>
    );
};

const mapStateToProps = state => ({
    filterORFs: state.analyses.filterORFs,
    maxSequenceLength: getMaxSequenceLength(state),
    hit: getActiveHit(state)
});

export default connect(mapStateToProps)(NuVsDetail);
