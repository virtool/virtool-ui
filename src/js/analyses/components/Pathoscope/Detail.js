import { filter, map, sum } from "lodash-es";
import React from "react";
import { connect } from "react-redux";
import { ScrollSync } from "react-scroll-sync";
import styled from "styled-components";
import { getActiveHit } from "../../selectors";
import { PathoscopeIsolate } from "./Isolate";

const getContextValue = (isolates, onRendered) => {
    let count = sum(map(isolates, isolate => isolate.sequences.length));

    return {
        count,
        onRendered: () => {
            count--;
            if (count === 0) {
                onRendered();
            }
        }
    };
};

const StyledPathoscopeDetail = styled.div``;

export const PathoscopeDetailContext = React.createContext();

export const PathoscopeDetail = ({ filterIsolates, hit, mappedReads, showPathoscopeReads, onRendered }) => {
    const { isolates, pi } = hit;

    const filtered = filter(isolates, isolate => filterIsolates === false || isolate.pi >= 0.03 * pi);

    const contextValue = getContextValue(filtered, onRendered);

    const isolateComponents = map(filtered, isolate => (
        <PathoscopeIsolate
            key={isolate.id}
            {...isolate}
            reads={Math.round(isolate.pi * mappedReads)}
            showPathoscopeReads={showPathoscopeReads}
        />
    ));

    return (
        <PathoscopeDetailContext.Provider value={contextValue}>
            <StyledPathoscopeDetail>
                <ScrollSync>
                    <div>{isolateComponents}</div>
                </ScrollSync>
            </StyledPathoscopeDetail>
        </PathoscopeDetailContext.Provider>
    );
};

const mapStateToProps = state => ({
    hit: getActiveHit(state),
    filterIsolates: state.analyses.filterIsolates,
    mappedReads: state.analyses.detail.read_count,
    showPathoscopeReads: state.analyses.showPathoscopeReads
});

export default connect(mapStateToProps)(PathoscopeDetail);
