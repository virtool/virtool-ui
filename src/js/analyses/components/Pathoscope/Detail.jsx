import { filter, map, sum } from "lodash-es";
import React from "react";
import { connect } from "react-redux";
import { ScrollSync } from "react-scroll-sync";
import { getActiveHit } from "../../selectors";
import { PathoscopeIsolate } from "./Isolate";

function getContextValue(isolates, onRendered) {
    let count = sum(map(isolates, isolate => isolate.sequences.length));

    return {
        count,
        onRendered: () => {
            count--;
            if (count === 0) {
                onRendered();
            }
        },
    };
}
export const PathoscopeDetailContext = React.createContext();

export const PathoscopeDetail = ({ filterIsolates, hit, mappedReads, showPathoscopeReads, onRendered }) => {
    const { isolates, pi } = hit;

    const filtered = filter(isolates, isolate => filterIsolates === false || isolate.pi >= 0.03 * pi);

    const contextValue = getContextValue(filtered, onRendered);

    const isolateComponents = map(filtered, isolate => (
        <PathoscopeIsolate
            key={isolate.id}
            {...isolate}
            reads={Math.round(isolate.pi * mappedReads)}
            showPathoscopeReads={showPathoscopeReads}
        />
    ));

    return (
        <PathoscopeDetailContext.Provider value={contextValue}>
            <div>
                <ScrollSync>
                    <div>{isolateComponents}</div>
                </ScrollSync>
            </div>
        </PathoscopeDetailContext.Provider>
    );
};

function mapStateToProps(state) {
    return {
        hit: getActiveHit(state),
        filterIsolates: state.analyses.filterIsolates,
        mappedReads: state.analyses.detail.read_count,
        showPathoscopeReads: state.analyses.showPathoscopeReads,
    };
}

export default connect(mapStateToProps)(PathoscopeDetail);
