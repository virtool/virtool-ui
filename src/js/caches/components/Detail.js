import React, { useEffect } from "react";
import { connect } from "react-redux";
import styled from "styled-components";
import {
    LoadingPlaceholder,
    RelativeTime,
    SubviewHeader,
    SubviewHeaderAttribution,
    SubviewHeaderTitle
} from "../../base";
import { getCache } from "../actions";
import CacheQuality from "./Quality";
import CacheGeneral from "./General";
import CacheParameters from "./Parameters";

const StyledCacheDetail = styled.div`
    overflow-x: hidden;
`;

export const CacheDetail = ({ detail, match, sampleName, onGet }) => {
    useEffect(() => onGet(match.params.cacheId), [match.params.cacheId]);

    if (detail === null) {
        return <LoadingPlaceholder />;
    }

    return (
        <StyledCacheDetail>
            <SubviewHeader>
                <SubviewHeaderTitle>Cache for {sampleName}</SubviewHeaderTitle>
                <SubviewHeaderAttribution>
                    Created <RelativeTime time={detail.created_at} />
                </SubviewHeaderAttribution>
            </SubviewHeader>
            <CacheGeneral {...detail} />
            <CacheParameters parameters={detail.parameters} />
            <CacheQuality />
        </StyledCacheDetail>
    );
};

export const mapStateToProps = state => {
    return {
        detail: state.caches.detail,
        sampleName: state.samples.detail.name
    };
};

export const mapDispatchToProps = dispatch => {
    return {
        onGet: cacheId => {
            dispatch(getCache(cacheId));
        }
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(CacheDetail);
