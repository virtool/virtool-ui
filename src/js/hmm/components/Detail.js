import { get, map } from "lodash-es";
import React from "react";
import { connect } from "react-redux";
import styled from "styled-components";
import { getBorder } from "../../app/theme";

import {
    device,
    Badge,
    BoxGroup,
    BoxGroupHeader,
    Label,
    LoadingPlaceholder,
    NotFound,
    Table,
    ViewHeader,
    ViewHeaderTitle
} from "../../base";
import { getHmm } from "../actions";
import { ClusterMember } from "./ClusterMember";
import { HMMTaxonomy } from "./Taxonomy";

const ClusterTable = styled(Table)`
    border: none;
    display: flex;
    flex-flow: column;
    height: 330px !important;
    margin: 0;
    width: 100%;

    thead {
        flex: 0 0 auto;
        width: calc(100% - 0.9em);

        th {
            border-bottom: none;
        }
    }

    tbody {
        flex: 1 1 auto;
        display: block;
        overflow-y: auto;
        border-top: ${getBorder};

        tr {
            width: 100%;
        }

        tr:first-child td {
            border-top: none;
        }
    }

    thead,
    tbody tr {
        display: table;
        table-layout: fixed;
    }
`;

const TaxonomyGrid = styled.div`
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-column-gap: ${props => props.theme.gap.column};

    @media (max-width: ${device.tablet}) {
        grid-template-columns: 1fr;
    }
`;

export class HMMDetail extends React.Component {
    componentDidMount() {
        this.props.onGet(this.props.match.params.hmmId);
    }

    render() {
        if (this.props.error) {
            return <NotFound />;
        }

        if (this.props.detail === null) {
            return <LoadingPlaceholder margin="130px" />;
        }
        const clusterMembers = map(this.props.detail.entries, ({ name, accession, organism }, index) => (
            <ClusterMember name={name} accession={accession} organism={organism} key={index} />
        ));

        const names = map(this.props.detail.names, (name, index) => (
            <Label key={index} spaced>
                {name}
            </Label>
        ));

        const title = this.props.detail.names[0];

        return (
            <div>
                <ViewHeader title={title}>
                    <ViewHeaderTitle>{title}</ViewHeaderTitle>
                </ViewHeader>

                <BoxGroup>
                    <BoxGroupHeader>
                        <h2>General</h2>
                    </BoxGroupHeader>

                    <Table>
                        <tbody>
                            <tr>
                                <th>Cluster</th>
                                <td>{this.props.detail.cluster}</td>
                            </tr>

                            <tr>
                                <th>Best Definitions</th>
                                <td>{names}</td>
                            </tr>

                            <tr>
                                <th>Length</th>
                                <td>{this.props.detail.length}</td>
                            </tr>

                            <tr>
                                <th>Mean Entropy</th>
                                <td>{this.props.detail.mean_entropy}</td>
                            </tr>

                            <tr>
                                <th>Total Entropy</th>
                                <td>{this.props.detail.total_entropy}</td>
                            </tr>
                        </tbody>
                    </Table>
                </BoxGroup>

                <BoxGroup>
                    <BoxGroupHeader>
                        <h2>
                            Cluster Members <Badge>{this.props.detail.entries.length}</Badge>
                        </h2>
                    </BoxGroupHeader>
                    <ClusterTable>
                        <thead>
                            <tr>
                                <th>Accession</th>
                                <th>Name</th>
                                <th>Organism</th>
                            </tr>
                        </thead>
                        <tbody>{clusterMembers}</tbody>
                    </ClusterTable>
                </BoxGroup>

                <TaxonomyGrid>
                    <HMMTaxonomy title="Families" counts={this.props.detail.families} />
                    <HMMTaxonomy title="Genera" counts={this.props.detail.genera} />
                </TaxonomyGrid>
            </div>
        );
    }
}

export const mapStateToProps = state => ({
    error: get(state, "errors.GET_HMM_ERROR", null),
    detail: state.hmms.detail
});

export const mapDispatchToProps = dispatch => ({
    onGet: hmmId => {
        dispatch(getHmm(hmmId));
    }
});

export default connect(mapStateToProps, mapDispatchToProps)(HMMDetail);
