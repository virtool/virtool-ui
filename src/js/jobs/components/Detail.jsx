import { push } from "connected-react-router";
import { get } from "lodash-es";
import React from "react";
import { connect } from "react-redux";
import styled from "styled-components";
import { checkAdminRoleOrPermission } from "../../administration/utils";
import {
    Badge,
    ContainerNarrow,
    LoadingPlaceholder,
    NotFound,
    ViewHeader,
    ViewHeaderAttribution,
    ViewHeaderIcons,
    ViewHeaderTitle,
} from "../../base";
import { getWorkflowDisplayName } from "../../utils/utils";
import { archiveJob, cancelJob, getJob } from "../actions";
import JobError from "./Error";
import { JobAction } from "./Item/Action";
import { JobArgs } from "./JobArgs";
import JobSteps from "./Steps";

const JobDetailBadge = styled(Badge)`
    text-transform: capitalize;
`;

class JobDetail extends React.Component {
    componentDidMount() {
        this.props.getDetail(this.props.match.params.jobId);
    }

    handleClick = () => {
        this.props.onRemove(this.props.detail.id);
    };

    render() {
        if (this.props.error) {
            return <NotFound />;
        }

        if (this.props.detail === null) {
            return <LoadingPlaceholder />;
        }
        const detail = this.props.detail;

        const latest = detail.status[detail.status.length - 1];

        let color = "green";

        if (latest.state === "running") {
            color = "blue";
        }

        if (latest.state === "error" || latest.state === "cancelled" || latest.state === "timeout" || latest.state === "terminated") {
            color = "red";
        }

        const workflow = getWorkflowDisplayName(detail.workflow);

        return (
            <ContainerNarrow>
                <ViewHeader title={workflow}>
                    <ViewHeaderTitle>
                        {workflow} <JobDetailBadge color={color}>{latest.state}</JobDetailBadge>
                        <ViewHeaderIcons>
                            <JobAction
                                state={detail.state}
                                onCancel={() => this.props.onCancel(this.props.detail.id)}
                                onArchive={() => this.props.onArchive(this.props.detail.id)}
                                canCancel={this.props.canCancel}
                                canArchive={this.props.canArchive}
                            />
                        </ViewHeaderIcons>
                    </ViewHeaderTitle>
                    <ViewHeaderAttribution time={detail.status[0].timestamp} user={detail.user.handle} />
                </ViewHeader>

                <JobArgs workflow={detail.workflow} args={detail.args} />

                <JobSteps />

                <JobError error={latest.error} />
            </ContainerNarrow>
        );
    }
}

const mapStateToProps = state => ({
    error: get(state, "errors.GET_JOB_ERROR", null),
    detail: state.jobs.detail,
    canCancel: checkAdminRoleOrPermission(state, "cancel_job"),
    canArchive: checkAdminRoleOrPermission(state, "remove_job"),
});

const mapDispatchToProps = dispatch => ({
    getDetail: jobId => {
        dispatch(getJob(jobId));
    },
    onCancel: jobId => {
        dispatch(cancelJob(jobId));
    },
    onArchive: jobId => {
        dispatch(archiveJob(jobId));
        dispatch(push("/jobs"));
    },
});

export default connect(mapStateToProps, mapDispatchToProps)(JobDetail);
