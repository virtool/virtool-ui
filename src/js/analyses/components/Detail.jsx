import { get } from "lodash-es";
import React, { useEffect } from "react";
import { connect } from "react-redux";
import styled from "styled-components";
import {
    Box,
    Icon,
    LoadingPlaceholder,
    NotFound,
    RelativeTime,
    SubviewHeader,
    SubviewHeaderAttribution,
    SubviewHeaderTitle,
} from "../../base/index";
import { getWorkflowDisplayName } from "../../utils/utils";
import { getAnalysis } from "../actions";
import AODPViewer from "./AODP/Viewer";
import AnalysisCache from "./CacheLink";
import NuVsViewer from "./NuVs/Viewer";
import { PathoscopeViewer } from "./Pathoscope/Viewer";

const UnsupportedAnalysis = styled(Box)`
    display: flex;
    justify-content: center;
    align-items: center;

    i.fas {
        margin-right: 5px;
    }
`;

export function AnalysisDetail({ detail, error, match, sampleName, onGetAnalysis }) {
    const analysisId = match.params.analysisId;

    useEffect(() => {
        onGetAnalysis(analysisId);
    }, [analysisId]);

    if (error) {
        return <NotFound />;
    }

    if (detail === null || detail.id !== analysisId) {
        return <LoadingPlaceholder />;
    }

    if (!detail.ready) {
        return (
            <Box>
                <LoadingPlaceholder message="Analysis in progress" margin="1.2rem" />
            </Box>
        );
    }

    let content;

    if (detail.workflow === "pathoscope_bowtie") {
        content = <PathoscopeViewer />;
    } else if (detail.workflow === "nuvs") {
        content = <NuVsViewer />;
    } else if (detail.workflow === "aodp") {
        content = <AODPViewer />;
    } else {
        return (
            <>
                <UnsupportedAnalysis>
                    <Icon name={"info-circle"} />
                    Workflow not yet supported.
                </UnsupportedAnalysis>
            </>
        );
    }

    return (
        <div>
            <SubviewHeader>
                <SubviewHeaderTitle>
                    {getWorkflowDisplayName(detail.workflow)} for {sampleName}
                    <AnalysisCache />
                </SubviewHeaderTitle>
                <SubviewHeaderAttribution>
                    {detail.user.handle} started <RelativeTime time={detail.created_at} />
                </SubviewHeaderAttribution>
            </SubviewHeader>

            {content}
        </div>
    );
}

export function mapStateToProps(state) {
    return {
        detail: state.analyses.detail,
        error: get(state, "errors.GET_ANALYSIS_ERROR", null),
        sampleName: state.samples.detail.name,
    };
}

export function mapDispatchToProps(dispatch) {
    return {
        onGetAnalysis: analysisId => {
            dispatch(getAnalysis(analysisId));
        },
    };
}

export default connect(mapStateToProps, mapDispatchToProps)(AnalysisDetail);
