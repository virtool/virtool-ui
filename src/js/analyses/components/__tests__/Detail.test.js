import { AnalysisDetail, mapStateToProps } from "../Detail";

describe("<AnalysisDetail />", () => {
    let props;

    beforeEach(() => {
        props = {
            detail: {
                id: "foo",
                created_at: "2019-05-28T19:04:25.201000Z",
                ready: true,
                user: {
                    id: "bob"
                },
                workflow: "pathoscope_bowtie"
            },
            error: null,
            quality: {
                read_count: 1432
            },
            sampleName: "bar",
            match: {
                params: {
                    analysisId: "foo"
                }
            }
        };
    });

    it("should render", () => {
        const wrapper = shallow(<AnalysisDetail {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should render message when error is not null", () => {
        props = {
            ...props,
            detail: null,
            error: {
                id: "test_error"
            }
        };
        const wrapper = shallow(<AnalysisDetail {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should render loading when [detail=null]", () => {
        props.detail = null;
        const wrapper = shallow(<AnalysisDetail {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should render pending when [detail.ready=false]", () => {
        props.detail.ready = false;
        const wrapper = shallow(<AnalysisDetail {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should render expected view when workflow is NuVs]", () => {
        props.detail.workflow = "nuvs";
        const wrapper = shallow(<AnalysisDetail {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it("should render error when workflow is invalid]", () => {
        props.detail.workflow = "baz";
        const wrapper = shallow(<AnalysisDetail {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});

describe("mapStateToProps", () => {
    let state;
    let expected;

    beforeEach(() => {
        state = {
            analyses: {
                detail: {
                    id: "foo"
                }
            },
            errors: {
                GET_SAMPLE_ERROR: {
                    id: "test_error"
                }
            },
            samples: {
                detail: {
                    name: "Baz",
                    quality: { read_count: 1231 }
                }
            }
        };

        expected = {
            detail: {
                id: "foo"
            },
            error: null,
            quality: {
                read_count: 1231
            },
            sampleName: "Baz"
        };
    });

    it("should return props", () => {
        const props = mapStateToProps(state);
        expect(props).toEqual(expected);
    });

    it("should return error when it exists", () => {
        const error = {
            id: "test_get_analysis_error"
        };

        state.errors.GET_ANALYSIS_ERROR = error;

        const props = mapStateToProps(state);

        expect(props).toEqual({
            ...expected,
            error
        });
    });
});
