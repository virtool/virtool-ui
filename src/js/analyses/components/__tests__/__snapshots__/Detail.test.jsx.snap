// Vitest Snapshot v1

exports[`<AnalysisDetail /> > should render 1`] = `
<div>
  <SubviewHeader>
    <SubviewHeaderTitle>
      Pathoscope
       for 
      bar
      <Connect(AnalysisCacheLink) />
    </SubviewHeaderTitle>
    <SubviewHeaderAttribution>
       started 
      <RelativeTime
        time="2019-05-28T19:04:25.201000Z"
      />
    </SubviewHeaderAttribution>
  </SubviewHeader>
  <PathoscopeViewer />
</div>
`;

exports[`<AnalysisDetail /> > should render error when workflow is invalid] 1`] = `
<div>
  Invalid analysis content
</div>
`;

exports[`<AnalysisDetail /> > should render expected view when workflow is NuVs] 1`] = `
<div>
  <SubviewHeader>
    <SubviewHeaderTitle>
      NuVs
       for 
      bar
      <Connect(AnalysisCacheLink) />
    </SubviewHeaderTitle>
    <SubviewHeaderAttribution>
       started 
      <RelativeTime
        time="2019-05-28T19:04:25.201000Z"
      />
    </SubviewHeaderAttribution>
  </SubviewHeader>
  <NuVsViewer />
</div>
`;

exports[`<AnalysisDetail /> > should render loading when [detail=null] 1`] = `<LoadingPlaceholder />`;

exports[`<AnalysisDetail /> > should render message when error is not null 1`] = `<NotFound />`;

exports[`<AnalysisDetail /> > should render pending when [detail.ready=false] 1`] = `
<Box>
  <LoadingPlaceholder
    margin="1.2rem"
    message="Analysis in progress"
  />
</Box>
`;
